src/main/java/com/gmail/nossr50/util/Misc.java

```diff
@@ -128,6 +128,20 @@ public static void spawnItemsFromCollection(@Nullable Player player, @NotNull Lo
         }
     }
 
+    /**
+     * Drops the item from the item stack only if it is a sapling (or equivalent)
+     * Needed for TreeFeller
+     */
+    public static void spawnItemIfSapling(@NotNull Player player, @NotNull Location location, @NotNull Collection < ItemStack > drops, @NotNull ItemSpawnReason itemSpawnReason) {
+        String dropString;
```

Comment:
You should move this variable declaration to the same line it is initialized, you can also make it ```final```

Reply:
The code snippet shown doesn't have a variable declaration shown.   
If you mean dropString, it is reset inside the for loop, so declaring it where it is initialized would require the variable to be redone every loop, which is why I declared outside the loop and why it isn't final, so I'm guessing you meant to link different code, but I'm not sure what else you might mean to fix it?

Reply:
Ah, here I see you do mean drop string.  I didn't move it because it needs to be reset as the drops are looped.

Reply:
variable removed entirely now that using predicate